module_macaddress.psm1

```diff
@@ -0,0 +1,102 @@
+<#
+.SYNOPSIS
+    A simple script to retrieve MAC Address vendor by supplying MAC address
+.DESCRIPTION
+    This script downloads wireshark's Vendors Mac Address List from their gitweb.
+    This script has two functions in it:
+        * Update-MacAddressVendor
+        * Get-MacAddressVendor
+
+    Update-MacAddressVendor downloads the current vendor file and places it in a given directory, 
+    determined by the $DocumentPath variable.
+
+    Get-MacAddressVendor searches through the downloaded file for the three first octets of the MAC Address, 
+    and hopefully retrieves the correct vendor of it. 
+
+.EXAMPLE
+    # Update Vendor List
+    Update-MacAddressVendor
+
+    # Search through vendor list
+    Get-MacAddressVendor -MacAddress '00-50-56-C0-00-00'
+
+    ===================RESULT===================
+    00:50:56	Vmware	VMware, Inc.
+    ============================================
+    
+
+.NOTES
+    Script written by CodeBarbarian
+#>
+
+
+
+# Incase of a proxy
+[System.Net.WebRequest]::DefaultWebProxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
+
+
+################################# Script Config ##################################
+# -------------------------- Script Dependant Variables -------------------------#
+# Document Path - Where the vendor mac addresses textfile will be located
+$DocumentPath = ("c:$($ENV:HOMEPATH)\desktop")
+##################################################################################
+
+
+
+function Update-MacAddressVendor {
+    # Path to the wireshark repository
+    $APIUrl = 'https://code.wireshark.org/review/gitweb?p=wireshark.git;a=blob_plain;f=manuf;hb=HEAD'
+
+    $WebClient = New-Object System.Net.WebClient
+    $WebClient.DownloadFile($APIUrl, "$($DocumentPath)\vendor_macaddresses.txt")
+
+    Write-Host ("Update of MAC vendor list complete.") -ForegroundColor Green
+}
+
+function Get-MacAddressVendor {
+    param (
+        [parameter(mandatory=$true)]
+        [string]$MacAddress
+    ) 
+
+    # Simple check for this one
+    if ([String]::IsNullOrWhiteSpace($MacAddress)) {
+        Write-Host ("The mac address can not be null or whitespace") -ForegroundColor RED

```

Comment:
Please use the attribute `[ValidateNotNullOrEmpty()]` on the `$MacAddress` instead and add `[CmdletBinding()]`  (this will make it an 'advanded function') above the `param` line to enable the usage of this attribute. This will take care of it automatically. If you still want to have the check for WhiteSpace, then you should use `Write-Error` but I think the null or empty check is sufficient.

Reply:
Yep, that is correct. the problem lies in the environment copying the mac address from. Can co